skcosmo/sample_selection/simple_fps.py

```diff
@@ -0,0 +1,144 @@
+                 samples. once a sample is selected, the distance is not updated;
+                 the final list will reflect the distances when selected.
+    n_samples_to_select : int
+        The number of samples that were selected.
+
+    norms_ : ndarray of shape (n_samples,)
+        The self-covariances of each of the samples
+
+    X_selected_ : ndarray (n_samples_to_select, n_features)
+                  The samples selected
+
+    y_selected_ : ndarray (n_samples_to_select, n_properties)
+                  The corresponding target values for the samples selected
+
+    eligible_ : ndarray of shape (n_samples,), dtype=bool
+                A mask of samples eligible for selection
+
+    n_selected_ : int
+                The number of samples that have been selected thus far
+
+    report_progress : callable
+                A wrapper to report the progress of the selector using a `tqdm` style
+                progress bar
+
+    score_threshold : float (optional)
+                A score below which to stop selecting samples
+
+    selected_idx_ : ndarray of integers
+                    indices of the selected samples, with respect to the
+                    original fitted matrix
+
+    support_ : ndarray of shape (n_samples,), dtype=bool
+        The mask of selected samples.
+
+    """
+
+    def __init__(
+        self,
+        n_samples_to_select=None,
+        initialize=0,
+        tolerance=1e-12,
+        progress_bar=False,
+    ):
+
+        scoring = self.score
+        self.initialize = initialize
+
+        super().__init__(
+            scoring=scoring,
+            n_samples_to_select=n_samples_to_select,
+            progress_bar=progress_bar,
+            score_thresh_to_select=tolerance,
+        )
+
+    def _calculate_distances(self, X, last_selected):
+        return self.norms_ + self.norms_[last_selected] - 2 * X[last_selected] @ X.T
+
+    def _set_norms(self, X, y):
+        self.norms_ = (X ** 2).sum(axis=1)
+
+    def _init_greedy_search(self, X, y, n_to_select):
+        """
+        Initializes the search. Prepares an array to store the selected
+        samples, selects the initial sample (unless provided), and
+        computes the starting haussdorf distances.
+        """
+
+        super()._init_greedy_search(X, y, n_to_select)
+        self._set_norms(X, y)
+
+        if self.initialize == "random":
+            initialize = np.random.randint(X.shape[0])
+        elif isinstance(self.initialize, numbers.Integral):
+            initialize = self.initialize
+        else:
+            raise ValueError("Invalid value of the initialize parameter")
+
+        self.selected_idx_[0] = initialize
+        self.haussdorf_ = np.full(X.shape[0], np.inf)
+        self.haussdorf_at_select_ = np.full(X.shape[0], np.inf)
+        self._update_post_selection(X, y, self.selected_idx_[0])
+
+    def score(self, X, y):
+        """ Returns the Haussdorf distances of all samples to previous selections"""
```

Comment:
Since `score` takes parameters, even though it ignores them, this should probably be stated in the docstring
```suggestion
        """ 
        Returns the Haussdorf distances of all samples to previous selections
        
        Parameters
        ----------
        X : ignored
        y : ignored
        
        Returns
        -------
        haussdorf : Haussdorf distances
        """
```